app/features/build/views/build.erb

```diff
@@ -34,11 +34,34 @@
 
 <hr />
 
+<% if build.artifacts.count > 0 %>
+  <h1>Artifacts</h1>
+  <table class="mdl-data-table mdl-js-data-table mdl-shadow--2dp">
+    <thead>
+      <tr>
+        <tr>
+          <th>Artifact type</th>
+          <th>Path</th>
+        </tr>
+      </tr>
+    </thead>
+    <tbody>
+      <% build.artifacts.each do |current_artifact| %>
+        <tr>
+          <td><%= current_artifact.type %></td>
+          <td><%= current_artifact.reference %></td>
```

Comment:
Same, use `retrieve!` method if possible, maybe I should limit the readability scope of the `reference` attribute

Reply:
Yep